### PR TITLE
ARC-1477 - Replace the header on Github Config Page

### PR DESCRIPTION
**Pull request by krazziekay**
**Changes**
- The `username-logout` options in the header for GitHub Pages(github/configuration and subscriptions) were replaced by `partial: navigation`.
- The logic for go-back in `navigation` was updated.
  - If there is no `previous-path-page`, then do a history.back().
  - Added a new property `close-on-back`. If this is true, then close the current browser window.

**How was this tested**
- This was tested in staging.  

**Screenshot**
image
image

### CODE REVIEW

File: static/js/navigation.js [hunk 1]
```diff
@@ -2,5 +2,15 @@
...
+	const closeOnBack = $(event.target).data("close-on-back");
+
+	if (closeOnBack) {
+		window.close();
+	} else {
+		if (previousPagePath) {
+			AP.navigator.go('addonmodule', { moduleKey: previousPagePath });
+		} else {
+			history.back();
+		}
```
Comment (joshkay10) [inline static/js/navigation.js:14]: /nit reduce nesting, could return early or change the `else { if {` to a `else if`

```
if (closeOnBack) {
 window.close();
 return;
}
if (previousPagePath) {
 AP.navigator.go('addonmodule', { moduleKey: previousPagePath });
 return;
}
history.back();
```

```
if (closeOnBack) {
 window.close();
} else if (previousPagePath) {
 AP.navigator.go('addonmodule', { moduleKey: previousPagePath });
} else {
 history.back();
}
```

File: views/partials/navigation.hbs [hunk 2]
```diff
@@ -1,6 +1,10 @@
 <nav class="navigation">
     <button class="navigation__backButton__container navigation__link go-back">
-        <img class="navigation__icon" src="/public/assets/arrow-left.svg" alt="Go back to {{previousPageName}}" data-previous-page-path="{{ previousPagePath }}" />
+        <img class="navigation__icon" src="/public/assets/arrow-left.svg"
+             alt="Go back to {{previousPageName}}"
+             data-previous-page-path="{{ previousPagePath }}"
+             data-close-on-back="{{ closeOnBack }}"
```
Comment (joshkay10) [inline views/partials/navigation.hbs:6]: //super nit.... but since were here, remove the extra spaces `{{ closeOnBack }}{{ closeOnBack }}`

Comment (krazziekay) [reply views/partials/navigation.hbs]: Removed the space

File: static/js/navigation.js (hunk 1 above)
Comment (krazziekay) [reply static/js/navigation.js]: Updated by using `if-elseif-else`

**Review by joshkay10 — DISMISSED** [review]

**Review by mboudreau — APPROVED** [review]

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**